packages/ui-react/src/components/Sidebar/Sidebar.tsx

```diff
@@ -18,16 +19,62 @@ const Sidebar: React.FC<SidebarProps> = ({ isOpen, onClose, children }) => {
   const { t } = useTranslation('menu');
   const lastFocusedElementRef = useRef<HTMLElement | null>(null);
   const sidebarRef = useRef<HTMLDivElement>(null);
+  const [visible, setVisible] = useState(false);
+
   const htmlAttributes = { inert: !isOpen ? '' : undefined }; // inert is not yet officially supported in react. see: https://github.com/facebook/react/pull/24730
 
   useEffect(() => {
     if (isOpen) {
+      // Before inert on the body is applied in Layout, we need to set this ref
       lastFocusedElementRef.current = document.activeElement as HTMLElement;
+
+      // When opened, adjust the margin-right to accommodate for the scrollbar width to prevent UI shifts in background
+      document.body.style.marginRight = `${scrollbarSize()}px`;
+      document.body.style.overflowY = 'hidden';
+
+      // Scroll the sidebar to the top if the user has previously scrolled down in the sidebar
+      if (sidebarRef.current) {
+        sidebarRef.current.scrollTop = 0;
+      }
+    } else {
+      document.body.style.removeProperty('margin-right');
+      document.body.style.removeProperty('overflow-y');
+    }
+  }, [isOpen]);
+
+  useEffect(() => {
+    const handleEscKey = (event: KeyboardEvent) => {
+      if (isOpen && event.key === 'Escape') {
```

Comment:
We aligned the sidebar behavior with the modal component because it essentially also presents a modal. When doing so, it is expected to support certain keyboard shortcuts. Although it might not be needed for mobile and tablets, it doesn't hurt to have it there just in case someone uses assertive technology on these specific devices or uses 200% zoom:

**1280px viewport with 200% zoom**
![image](https://github.com/jwplayer/ott-web-app/assets/3996119/af4e9864-c4ed-47d7-afe8-e4985595cac7)